sklearn/tree/_splitter.pyx

```diff
@@ -481,7 +481,7 @@ cdef class BestSplitter(BaseDenseSplitter):
 
                             if current_proxy_improvement > best_proxy_improvement:
                                 best_proxy_improvement = current_proxy_improvement
-                                current.threshold = (Xf[p - 1] + Xf[p]) / 2.0
+                                current.threshold = Xf[p - 1] / 2.0 + Xf[p] / 2.0
```

Comment:
This deserves comment if you don't want someone to come and try change it back in the future... (though of course the test will fail.)